### ISSUE TITLE
Support encryption

### ISSUE THREAD

**Issue by KirillovDenis** [nspcc-dev/neofs-s3-gw#193]
Should s3-gw support [encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingServerSideEncryption.html)?

**Comment by realloc** [comment]
Yes, but we need to figure out the details. I doubt it could work properly in public network, so we could start with some private|local key management services integration. Let's wait a little to survey the possible use cases.

**Comment by alexvanin** [comment]
As for public networks, encryption should be done completely on client side, before object upload.

**Comment by alexvanin** [comment]
Consider having KMS service and binding it with S3 Gateway.

**Comment by roman-khimov** [comment]
Requires secret service.
